File: New BackEnd/demo/src/main/java/com/example/demo/models/Company.java
package com.example.demo.models;


import ch.qos.logback.core.net.server.Client;
import jakarta.persistence.*;
import lombok.Data;

import java.util.List;

@Data
@Entity
@Table(name = "company")
public class Company {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String address;
    private String contact;
    @OneToOne(mappedBy = "company")
    private Contactor contactor;
    @OneToMany(mappedBy = "company")
    private List<Worker> workers;
    @OneToMany(mappedBy = "company")
    private List<Devis> devisList;
    @OneToMany(mappedBy = "company")
    private List<DevisRequest> devisRequestList;
    @OneToMany(mappedBy = "company")
    private List<SolarPanel> SolarePanelList;
    @OneToMany(mappedBy = "company")
    private List<Meter> meterList;
    @OneToMany(mappedBy = "company")
    private List<SystemFixing> systemFixingList;
    @OneToMany(mappedBy = "company")
    private List<Inverter> inverterList;
    @OneToMany(mappedBy = "company")
    private List<Supplier> supplierList;
    @OneToMany(mappedBy = "company")
    private List<Battery> batteryList;
    @OneToMany(mappedBy = "company")
    private List<Cable> cableList;
    @OneToMany(mappedBy = "company")
    private List<Customer> customerList;
    @OneToMany(mappedBy = "company")
    private List<Activity> activityList;
    @OneToMany(mappedBy = "company")
    private List<Construction> constructionList;
}

File: photovoltaicV2/src/app/services/models/company.ts
/* tslint:disable */
/* eslint-disable */
import { Activity } from '../models/activity';
import { Battery } from '../models/battery';
import { Cable } from '../models/cable';
import { Construction } from '../models/construction';
import { Contactor } from '../models/contactor';
import { Customer } from '../models/customer';
import { Devis } from '../models/devis';
import { DevisRequest } from '../models/devis-request';
import { Inverter } from '../models/inverter';
import { Meter } from '../models/meter';
import { SolarPanel } from '../models/solar-panel';
import { Supplier } from '../models/supplier';
import { SystemFixing } from '../models/system-fixing';
import { Worker } from '../models/worker';
export interface Company {
  activityList?: Array<Activity>;
  address?: string;
  batteryList?: Array<Battery>;
  cableList?: Array<Cable>;
  constructionList?: Array<Construction>;
  contact?: string;
  contactor?: Contactor;
  customerList?: Array<Customer>;
  devisList?: Array<Devis>;
  devisRequestList?: Array<DevisRequest>;
  id?: number;
  inverterList?: Array<Inverter>;
  meterList?: Array<Meter>;
  name?: string;
  solarePanelList?: Array<SolarPanel>;
  supplierList?: Array<Supplier>;
  systemFixingList?: Array<SystemFixing>;
  workers?: Array<Worker>;
}
